dxr/utils.py

```diff
@@ -199,6 +199,23 @@ def file_text(file_path):
     with open(file_path) as file:
         return file.read()
 
+def bucket(things, key):
+    """Return a map of key -> list of things."""
+    ret = defaultdict(list)
+    for thing in things:
+        ret[key(thing)].append(thing)
+    return ret
+
```

Comment:
Once we merge to master and know we won't need to do any merges out *from* master, we'll get a lightweight flake8 checker in place.